src/OrbitDB.js

```diff
@@ -295,10 +305,16 @@ class OrbitDB {
     // Create the database address
     const dbAddress = await this._determineAddress(name, type, options)
 
-    options.cache = this.caches[options.directory || 'default']
     if (!options.cache) {
-      const cacheStorage = await this.storage.createStore(options.directory)
-      this.caches[options.directory] = options.cache = new Cache(cacheStorage)
+      if (options.directory) {
+        const cacheStorage = await this.storage.createStore(options.directory)
+        this.caches[options.directory] = options.cache = new Cache(cacheStorage)
+        if (options.cache._store.db.status !== 'open') {
+          await options.cache._store.open()
```

Comment:
Can you help me understand why we need to check the store status directly here? It feels like this should be handled on another layer (in the store/storage modules?)...

Reply:
This is a safeguard against trying to access a store too soon, before it is open. I put the check here to prevent an API breaking change.

Since constructors are not async we would have to migreate to an `async Cache.create` function and have people update their code. This way we can avoid that.